lib/CL/devices/devices.c

```diff
@@ -431,9 +431,11 @@ pocl_init_devices()
             pocl_devices[dev_index].available = 0;
           }
 
-          if (dev_index == 0)
-            pocl_devices[dev_index].type |= CL_DEVICE_TYPE_DEFAULT;
-
+          if (pocl_get_bool_option ("POCL_WORKAROUND_DEVTYPE", 0) == 0)
```

Comment:
> "What bug?" people ask. 

The device type bug. It's literallly spelled out there. There isn't all that much bugs related to device-type :)

> POCL_NO_DEFAULT_DEVICE_TYPE_FLAG (that explicitly tells what it does)
Yep it's explicit. It just has one tiny problem, nobody will ever remember it. 

> I'm not sure if it's mandatory according to the specs to use/return the default device flag actually. 

It doesn't seem to be. We could remove it, but then we need to special case CL_DEVICE_TYPE_DEFAULT for clCreateContextFromType(). I can also remove the commit entirely (it's really just a matter of convenience).